src/main/java/dk/kb/labsapi/metadataFormats/IllustrationMetadata.java

```diff
@@ -24,48 +24,50 @@
  *     <li>x: height of illustration</li>
  * </ul>
  */
-public class IllustrationMetadata {
+public class IllustrationMetadata extends BasicMetadata {
     private static final Logger log = LoggerFactory.getLogger(IllustrationMetadata.class);
-    private String pageUUID;
-    private double pageWidth;
-    private double pageHeight;
     private String id;
     private double x;
     private double y;
     private double w;
     private double h;
-    static final Pattern illustrationPattern = Pattern.compile("id=(\\S*),x=(\\d*),y=(\\d*),w=(\\d*),h=(\\d*),doms_aviser_page:uuid:(\\S*),(\\d*),(\\d*)");
+    static final Pattern oldIllustrationPattern = Pattern.compile("id=(\\S*),x=(\\d*),y=(\\d*),w=(\\d*),h=(\\d*),doms_aviser_page:uuid:(\\S*),(\\d*),(\\d*)");
+    static final Pattern singleIllustrationPattern = Pattern.compile("id=(\\S*),x=(\\d*),y=(\\d*),w=(\\d*),h=(\\d*)");
 
-    public IllustrationMetadata(){
-    }
 
-    // Setter
-    public void setData(String input) {
-        Matcher m = illustrationPattern.matcher(input);
+    public IllustrationMetadata(String input){
+        Matcher m = oldIllustrationPattern.matcher(input);
         if (m.matches()) {
             this.id = m.group(1);
             this.x = Double.parseDouble(m.group(2));
             this.y = Double.parseDouble(m.group(3));
             this.w = Double.parseDouble(m.group(4));
             this.h = Double.parseDouble(m.group(5));
             this.pageUUID = m.group(6);
-            this.pageWidth = convertPixelsToInch1200(Integer.parseInt(m.group(7)));
-            this.pageHeight = convertPixelsToInch1200(Integer.parseInt(m.group(8)));
+            this.pageWidth = (long) convertPixelsToInch1200(Integer.parseInt(m.group(7)));
+            this.pageHeight = (long) convertPixelsToInch1200(Integer.parseInt(m.group(8)));
         } else {
             log.error("Regex matching failed. Please check that input and pattern aligns.");
         }
     }
 
-    // Getters
-    public String getPageUUID(){
-        return pageUUID;
-    }
-    public double getPageWidth() {
-        return pageWidth;
-    }
-    public double getPageHeight() {
-        return pageHeight;
+    public IllustrationMetadata(String illustrationString, String pageUUID, long pageWidth, long pageHeight){
+        Matcher m = singleIllustrationPattern.matcher(illustrationString);
+        if (m.matches()){
+            this.id = m.group(1);
+            this.x = Double.parseDouble(m.group(2));
+            this.y = Double.parseDouble(m.group(3));
+            this.w = Double.parseDouble(m.group(4));
+            this.h = Double.parseDouble(m.group(5));
+        } else {
+            log.error("Regex matching failed. Could not create IllustrationMetadata.");
```

Comment:
1. Remember to provide contextual information: What was the input that could not be matched?
2. `error` is reserved for issues threatening stability of the service. `warn` seems more appropriate her (ideally we would distinguish between operations of the service and the input & data themselves when logging, but in this project as well as in general, we do not do that)
3. The `IllustrationMetadata` seems to be in a sorry state if it is not matching the input. Shouldn't this throw an Exception or in some other way signal that later processing steps should not see it as valid?

Reply:
Forgot to log the problematic input :-)